### PR TITLE
feat: support recycle bin in content navigation #152

### PR DESCRIPTION
**Pull request by boyce-w**
**Summary**
Support Recycle Bin in the SAS Content Navigation Pane.
See details in the issue #152 

**Testing**
How did you test this change?
Test manually and run all the unit tests

### CODE REVIEW

**Comment by Zhirong2022** [comment]
1.Make sure there are no regression issues 
2.Verify that the file or folder will be moved to Recycle bin folder if performing delete action 
3.Verify that the file or folder will roll back to its original location if performing restore action
4.Verify that the file tab page will be closed at the same time if trying to delete a file in open status
5.Verify that the file is in read-only in Recycle bin folder
6.Verity that it will ask the user to Save, NotSave or Cancel if trying to delete a file in edit mode
7.Verify that the file will be handled properly if the user take Save, NotSave or Cancel action
7.1 "Save": The file will be saved and moved to Recycle bin folder
7.2 "NotSave" The file will not be saved and moved to Recycle bin folder
7.3 "Cancel" The popup confirmation message dialog close and the file keeps in edit mode
8.Verify that the file or folder can be deleted successfully in Recycle bin folder
9.Verify that it can Empty the Recycle bin folder successfully
10.Verify 1~9 in OS: Windows and Mac
11.Verify 1~10 in NLS environment

**Review by scottdover — COMMENTED** [review]
This is looking good so far. I appreciate you taking this one on :)

File: client/src/components/ContentNavigator/utils.ts [hunk 1]
```diff
@@ -53,30 +53,34 @@ export const resourceType = (item: ContentItem): string | undefined => {
...
     return;
   }
+  const { write, delete: del, addMember } = getPermission(item);
+  const isRecycled = isItemInRecycleBin(item);
+  const actions = [
+    addMember && !isRecycled ? "createChild" : undefined,
+    del ? "delete" : undefined,
+    write && !isRecycled ? "update" : undefined,
+    write && isRecycled ? "restore" : undefined,
+  ].filter((action) => !!action);
```
[the remark this answers is not in the document]
Comment (scottdover) [reply client/src/components/ContentNavigator/utils.ts:63]: This is all to say, you can leave your code as-is, or if you see a good way to make it a little more readable, that'd be awesome, too :)

File: client/src/components/ContentNavigator/index.ts [hunk 2]
```diff
@@ -85,6 +98,36 @@ class ContentNavigator {
...
+    );
+
+    commands.registerCommand(
+      "SAS.emptyRecycleBin",
+      async (resource: ContentItem) => {
+        const children = await this.contentDataProvider.getChildren(resource);
+        if (
+          !(await Promise.all(
+            children.map((child) =>
+              this.contentDataProvider.deleteResource(child)
```
[the remark this answers is not in the document]
Comment (boyce-w) [reply client/src/components/ContentNavigator/index.ts:122]: Yes, that's right

**Comment by boyce-w** [comment]
I don't know why after I rebased the PR and resolved the conflicts, this branch goes wrong. It contains the commits pushed in the main branch and the changed files are contaminated. So I created a new PR #190 and close this one. Sorry for confusing.

**Comment by Zhirong2022** [comment]
Since the issue is Dup to PR#190, verify it as well.

**Pull request closed without merging**